plasmapy/diagnostics/charged_particle_radiography.py

```diff
@@ -347,6 +352,9 @@ def add_wire_mesh(
 
         """
 
+        # Simulation has not run, because adding mesh changes the simulation
+        self._has_run = False
```

Comment:
Are there any other "things" that need to be "reset" when a user modifies `Tracker` (e.g. voiding/deleting results).  This might be a good thing to do in a `_reset_results` method.

Reply:
Not really -  in 99% of use cases I expect that the simulation will only be run once (since the overhead to instantiate a new simulation is so much lower than actually running the simulation). The only ways to access the results (other than going in and pulling out the attributes manually) are as output from `run()` or through `results_dict()` which now checks to make sure the results are current.

Reply:
I agree that a very few percentage of situation will re-run a simulation, but if it's possible to re-run then we need to properly handle it.  Otherwise, it could lead to unintended consequences.  For now, I would be okay with not allowing any modification to the `Tracker` object after the sim is run.

Reply:
Ok, done. 

I think this will be the permeant solution: I tried making it so you could re-run the simulation with changes but this necessitates storing copies of a lot of the arrays in a way that just isn't memory efficient for large calculations.